packages/postgresql/src/lib/PostgreSQLProvider.ts

```diff
@@ -506,60 +506,70 @@ export class PostgreSQLProvider<StoredValue = unknown> extends JoshProvider<Stor
   }
 
   public async [Method.Random](payload: Payload.Random<StoredValue>): Promise<Payload.Random<StoredValue>> {
-    const { count, duplicates } = payload;
+    const { count, unique } = payload;
     const size = await this.handler.size();
 
-    if (size === 0) return { ...payload, data: [] };
-    if (size < count) {
+    if (unique && size < count) {
       payload.errors.push(this.error({ identifier: CommonIdentifiers.InvalidCount, method: Method.Random }));
 
       return payload;
     }
 
+    if (size === 0) {
+      payload.errors.push(this.error({ identifier: CommonIdentifiers.MissingData, method: Method.Random }));
+
+      return payload;
+    }
+
     payload.data = [];
 
     const keys = await this.handler.keys();
 
-    if (duplicates) {
-      while (payload.data.length < count) {
-        const key = keys[Math.floor(Math.random() * size)];
-
-        payload.data.push((await this.handler.get(key))!);
-      }
-    } else {
+    if (unique) {
       const randomKeys = new Set<string>();
 
       while (randomKeys.size < count) randomKeys.add(keys[Math.floor(Math.random() * keys.length)]);
```

Comment:
Same as Maria, any cool database things we can do to make this more optimized?

Reply:
Seems like the same as maria, where it can generate a random number, but not efficently get random keys.
https://www.techonthenet.com/postgresql/functions/random.php